common/src/main/java/com/wynntils/gui/render/MapRenderer.java

```diff
@@ -99,10 +116,231 @@ public static void renderMapQuad(
+                                FontRenderer.TextShadow.NORMAL);
+            }
+        }
+
         // disable rotation if necessary
         if (followPlayerRotation) {
             poseStack.popPose();
         }
+
+        renderServicePois(
+                poseStack,
+                mapCenterX,
+                mapCenterZ,
+                centerX,
+                centerZ,
+                scale,
+                poiScale,
+                mouseCoordinates,
+                followPlayerRotation,
+                mapTopX,
+                mapLeftZ,
+                mapBottomX,
+                mapRightZ);
+    }
+
+    private static void renderServicePois(
+            PoseStack poseStack,
+            float mapCenterX,
+            float mapCenterZ,
+            float centerX,
+            float centerZ,
+            float scale,
+            float poiScale,
+            Pair<Integer, Integer> mouseCoordinates,
+            boolean followPlayerRotation,
+            float mapTopX,
+            float mapLeftZ,
+            float mapBottomX,
+            float mapRightZ) {
+        List<ServicePoi> servicePois = MapModel.getAllPois().stream()
+                .filter(poi -> {
+                    if (!(poi instanceof ServicePoi)) return false;
+
+                    return isPoiVisible(
+                            mapCenterX,
+                            mapCenterZ,
+                            centerX,
+                            centerZ,
+                            scale,
+                            mapTopX,
+                            mapLeftZ,
+                            mapBottomX,
+                            mapRightZ,
+                            poi);
+                })
+                .map(poi -> (ServicePoi) poi)
+                .toList();
+
+        if (mouseCoordinates != null) {
+            hovered = null;
+        }
+
+        for (ServicePoi servicePoi : servicePois) {
+            renderServicePoi(
+                    poseStack,
+                    mouseCoordinates,
+                    mapCenterX,
+                    mapCenterZ,
+                    centerX,
+                    centerZ,
+                    followPlayerRotation,
+                    scale,
+                    poiScale,
+                    servicePoi);
+        }
+
+        if (McUtils.mc().screen instanceof MainMapScreen mainMapScreen) {
+            mainMapScreen.setHovered(hovered);
+        }
+
+        if (hovered != null && mouseCoordinates != null) {
+            float textureXPosition = getRenderX(hovered, mapCenterX, centerX, 1f / scale);
+            float textureZPosition = getRenderZ(hovered, mapCenterZ, centerZ, 1f / scale);
+
+            float width = hovered.getIcon().width() * poiScale;
+            float height = hovered.getIcon().height() * poiScale;
+
+            poseStack.pushPose();
+
+            float renderX = textureXPosition - width / 2f;
+            float renderZ = textureZPosition - height / 2f;
```

Comment:
Same comment

Reply:
What? Alignment to what? This is so POIs are rendered centered to their coordinates. No config needed here.